Tests/TuistKitTests/Services/GenerateServiceTests.swift

```diff
@@ -0,0 +1,142 @@
+ import Basic
+ import Foundation
+ import TuistCore
+ import TuistLoader
+ import TuistSupport
+ import XcodeProj
+ import XCTest
+ @testable import TuistCoreTesting
+ @testable import TuistKit
+ @testable import TuistLoaderTesting
+ @testable import TuistSupportTesting
+
+ final class GenerateServiceTests: TuistUnitTestCase {
```

Comment:
I really like that we are extracting the logic from the commands into services. Nicely done @fortmarek